File: leetcode869/leetcode869.py
//https://leetcode.com/problems/reordered-power-of-2/description/

class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        count=self.counter(n)
        for i in range(30):
            if self.counter(1<<i)==count:
                return True
        return False
    def counter(self,n):
        count=0
        while n>0:
            count+=math.pow(10,n%10)
            n//=10
        return count

File: leetcode869/leetcode869.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        int count = counter(n);
        for (int i = 0; i < 30; ++i) {
            if (counter(1 << i) == count)
                return true;
        }
        return false;
    }
private:
    int counter(int n) {
        int count = 0;
        while (n > 0) {
            count += pow(10, n%10);
            n /= 10;
        }
        return count;
    }
};
